temporal/error.go

```diff
@@ -166,6 +168,88 @@ var (
 	ErrSkipScheduleUpdate = internal.ErrSkipScheduleUpdate
 )
 
+// Use instance of the interface to set all the desired attributes of a new ApplicationError
+// To get a new instance use ErrorAttributes function
+type ApplicationErrorAttributes interface {
```

Comment:
I don't think there's any value in this being an interface

Reply:
+1

Reply:
I think we want Chad's way of constructing application error, don't see a reason for including anything in this file other than exporting the internal constructor.